### PR TITLE
feat: mvp

### PR DESCRIPTION
**Pull request by Odrin**
## Status

**READY/IN DEVELOPMENT/HOLD**

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: example/lib/webview/view/webview_page.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+const address =
+    '0:727a540fb41fba5767e8fb5aaf0c9b9b0c9aa4ff8d534c45e5ba68742dacc134';
+const publicKey =
+    '9599d7a809bd0787b2dd995df6408bb0c25ea4c1cb9a26d83d68639797abb5e3';
+
+class WebviewPage extends StatelessWidget {
+  const WebviewPage({super.key});
+
+  dynamic _getProviderState() => {
```
Comment (nesquikm) [inline example/lib/webview/view/webview_page.dart]: I think Map<String, Object> will be better than just a dynamic.

File: example/lib/webview/view/webview_page.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        'version': '0.3.0',
+        'numericVersion': 3000,
+        'networkId': 1,
+        'selectedConnection': 'mainnet',
+        'supportedPermissions': ['basic', 'accountInteraction'],
+        'subscriptions': <String, Object>{},
+        'permissions': <String, Object>{},
+      };
+
+  dynamic _requestPermissions() => {
```
Comment (nesquikm) [inline example/lib/webview/view/webview_page.dart]: Map<String, Object> will be better.

**Comment by nesquikm** [comment]
I think we should drop "counter" feature (example/lib/counter)

File: example/lib/webview/view/webview_page.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    final l10n = context.l10n;
+    return Scaffold(
+      appBar: AppBar(title: Text(l10n.counterAppBarTitle)),
+      body: SafeArea(
+        child: InAppWebView(
+          onWebViewCreated: (InAppWebViewController controller) async {
+            await controller.initNekotonProvider(
+              handler: ({required String method, dynamic params}) {
+                switch (method) {
+                  case 'getProviderState':
```
Comment (nesquikm) [inline example/lib/webview/view/webview_page.dart]: It's better to make an enum and use it's name as string here. For example Methods.getProviderState.name.

File: .gitignore [hunk 4]
```diff
@@ -40,4 +40,6 @@ app.*.symbols
 app.*.map.json
 
 # Test related
-coverage
\ No newline at end of file
+coverage
+
+node_modules
```
Comment (nesquikm) [inline .gitignore]: No newline at eof.

File: example/android/build.gradle [hunk 5]
```diff
@@ -24,6 +24,6 @@ subprojects {
     project.evaluationDependsOn(':app')
 }
 
-task clean(type: Delete) {
+tasks.register("clean", Delete) {
```
Comment (nesquikm) [inline example/android/build.gradle:27]: I don't get it. Could you please comment on this?

File: lib/src/nekoton_webview.dart [hunk 6]
```diff
@@ -1,7 +1,47 @@
...
+      userScript: UserScript(
+        source: await rootBundle
+            .loadString('packages/nekoton_webview/assets/main.js'),
+        injectionTime: UserScriptInjectionTime.AT_DOCUMENT_START,
+      ),
+    );
+
+    addJavaScriptHandler(
+      handlerName: 'nekotonWebviewHandler',
+      callback: (List<dynamic> arguments) {
```
Comment (nesquikm) [inline lib/src/nekoton_webview.dart:32]: It's better to create a clear interface by providing an abstract class in the library and forcing user to pass an instance of inherited class here. Thus, the library user will have to implement the required methods with the necessary arguments, and also receive documentation for each of them (yeah, docs please?).

**Review by nesquikm — CHANGES_REQUESTED** [review]

**Review by Alex-A4 — APPROVED** [review]

File: melos.yaml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    failFast: true
+    packageFilters:
+      flutter: true
+      dirExists: integration_test
+
+  generate-js:
+    run: bash scripts/generate-js.sh
+    description: Generate JS bindings
+
+  generate-dart:
```
Comment (nesquikm) [inline melos.yaml:43]: Can we merge generate-dart with codegen?

**Review by nesquikm — APPROVED** [review]

**Pull request merged**